tests/dragonfly/replication_test.py

```diff
@@ -1541,3 +1542,76 @@ async def test_df_crash_on_memcached_error(df_local_factory):
         memcached_client.set(b"key", b"data", noreply=False)
 
     await c_master.close()
+
+
+@pytest.mark.asyncio
+@pytest.mark.slow
+async def test_network_disconnect(df_local_factory, df_seeder_factory):
+    replica = df_local_factory.create(port=BASE_PORT)
+    master = df_local_factory.create(port=BASE_PORT + 1)
+    seeder = df_seeder_factory.create(port=master.port)
+
+    df_local_factory.start_all([replica, master])
+    async with replica.client() as c_replica:
+        await seeder.run(target_deviation=0.1)
+
+        proxy = Proxy("localhost", BASE_PORT + 2, "localhost", master.port)
+        task = asyncio.create_task(proxy.start())
+
+        await c_replica.execute_command(f"REPLICAOF localhost {proxy.port}")
+
+        for _ in range(10):
+            await asyncio.sleep(random.randint(0, 10) / 10)
+            proxy.drop_connection()
+
+        # Give time to detect dropped connection and reconnect
+        await asyncio.sleep(0.5)
+        await wait_for_replica_status(c_replica, status="up")
+        await wait_available_async(c_replica)
+
+        capture = await seeder.capture()
+        assert await seeder.compare(capture, replica.port)
+        proxy.close()
+        try:
+            await task
+        except asyncio.exceptions.CancelledError:
+            pass
+
+
+@pytest.mark.asyncio
+@pytest.mark.slow
+async def test_network_disconnect_active_stream(df_local_factory, df_seeder_factory):
+    replica = df_local_factory.create(port=BASE_PORT, proactor_threads=4)
+    master = df_local_factory.create(port=BASE_PORT + 1, proactor_threads=4)
+    seeder = df_seeder_factory.create(port=master.port)
+
+    df_local_factory.start_all([replica, master])
+    async with replica.client() as c_replica:
+        await seeder.run(target_deviation=0.1)
+
+        proxy = Proxy("localhost", BASE_PORT + 2, "localhost", master.port)
+        task = asyncio.create_task(proxy.start())
+
+        await c_replica.execute_command(f"REPLICAOF localhost {proxy.port}")
+
+        fill_task = asyncio.create_task(seeder.run(target_ops=3000))
+
+        for _ in range(5):
+            await asyncio.sleep(random.randint(0, 10) / 5)
+            proxy.drop_connection()
+
+        seeder.stop()
+        await fill_task
+
+        # Give time to detect dropped connection and reconnect
+        await asyncio.sleep(0.5)
+        await wait_for_replica_status(c_replica, status="up")
+        await wait_available_async(c_replica)
```

Comment:
How do you check in the test that partial sync was done and not full sync?

Reply:
I didn't, now I do by checking the logs